File: pyevolve/envs/prima_vita/particle_class.py
import pygame
import random
import math
import os

from .global_constants import *


class Particle:
    def __init__(self):
        self.particleImg = pygame.image.load(
            os.path.join(os.path.dirname(__file__), "images/food.png")
        )
        self.particleX = random.randint(10, SCREEN_WIDTH - 10)
        self.particleY = random.randint(10, SCREEN_HEIGHT - 10)

    def show_particle(self):
        screen.blit(self.particleImg, (self.particleX, self.particleY))

    def show_close(self):  # testing function
        screen.blit(
            pygame.image.load(
                os.path.join(os.path.dirname(__file__), "images/food_near.png")
            ),
            (self.particleX, self.particleY),
        )

File: cpp/particle_class.cpp
#include <cstdlib>
#include <iostream>

#include "global_constants.hpp"

class Particle {
private:
  int particleX, particleY;

public:
  Particle() {
    particleX = 10 + (rand() % static_cast<int>(SCREEN_WIDTH - 20 + 1));
    particleY = 10 + (rand() % static_cast<int>(SCREEN_HEIGHT - 20 + 1));
  }

  int get_particleX() {
    return particleX;
  }

  int get_particleY() {
    return particleY;
  }
};

int main() {
  Particle particle;
  std::cout << particle.get_particleX() << std::endl;
  std::cout << particle.get_particleY() << std::endl;
}
